src/generators/legacy-json/utils/parseSignature.mjs

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+import { PARAM_EXPRESSION } from '../constants.mjs';
+
+const OPTIONAL_LEVEL_CHANGES = { '[': 1, ']': -1, ' ': 0 };
+
+/**
+ * @param {string} parameterName
+ * @param {number} optionalDepth
+ * @returns {[string, number, boolean]}
+ */
+function parseNameAndOptionalStatus(parameterName, optionalDepth) {
+  // Let's check if the parameter is optional & grab its name at the same time.
+  //  We need to see if there's any leading brackets in front of the parameter
+  //  name. While we're doing that, we can also get the index where the
+  //  parameter's name actually starts at.
+  let startingIdx = 0;
+  for (; startingIdx < parameterName.length; startingIdx++) {
+    const levelChange = OPTIONAL_LEVEL_CHANGES[parameterName[startingIdx]];
+
+    if (!levelChange) {
+      break;
+    }
+
+    optionalDepth += levelChange;
+  }
+
+  const isParameterOptional = optionalDepth > 0;
+
+  // Now let's check for any trailing brackets at the end of the parameter's
+  //  name. This will tell us where the parameter's name ends.
+  let endingIdx = parameterName.length - 1;
+  for (; endingIdx >= 0; endingIdx--) {
+    const levelChange = OPTIONAL_LEVEL_CHANGES[parameterName[endingIdx]];
+    if (!levelChange) {
+      break;
+    }
+
+    optionalDepth += levelChange;
+  }
+
+  return [
+    parameterName.substring(startingIdx, endingIdx + 1),
+    optionalDepth,
+    isParameterOptional,
+  ];
+}
+
+/**
+ * @param {string} parameterName
+ * @returns {[string, string | undefined]}
+ */
+function parseDefaultValue(parameterName) {
+  /**
+   * @type {string | undefined}
+   */
+  let defaultValue;
+
+  const equalSignPos = parameterName.indexOf('=');
```

Comment:
Nit: (and I believe this is applicable in many places): Add an empty line between assignment and usage :)